src/org/authInfo.ts

```diff
@@ -1065,6 +1070,30 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
     }
     throw new SfError(errorMsg);
   }
+
+  /**
+   * Returns `true` if the org is a Dev Hub.
+   *
+   * Check access to the ScratchOrgInfo object to determine if the org is a dev hub.
+   */
+  private async determineIfDevHub(instanceUrl: string, accessToken: string): Promise<boolean> {
+    // Make a REST call for the SratchOrgInfo obj directly.  Normally this is done via a connection
+    // but we don't want to create circular dependencies or lots of snowflakes
+    // within this file to support it.
+    const apiVersion = 'v51.0'; // hardcoding to v51.0 just for this call is okay.
+    const instance = ensure(instanceUrl);
+    const baseUrl = new SfdcUrl(instance);
+    const scratchOrgInfoUrl = `${baseUrl}/services/data/${apiVersion}/query?q=SELECT%20Id%20FROM%20ScratchOrgInfo%20limit%201`;
+    const headers = Object.assign({ Authorization: `Bearer ${accessToken}` }, SFDX_HTTP_HEADERS);
+
+    try {
+      await new Transport().httpRequest({ url: scratchOrgInfoUrl, method: 'GET', headers });
```

Comment:
The `Connection` class writes the authFile when grabing the latest api version:
https://github.com/forcedotcom/sfdx-core/blob/fa6e4e25255df57f2b80d05bfc62c38d192aa924/src/org/connection.ts#L502-L507

It's up to the consumer to save it or not with `.save()`, so I added this method that only hits the REST endpoint, it doesn't mutate/save the in-memory authConfig.